### ISSUE TITLE
Difference in starting Loss with and without Packing for Olmo3

### ISSUE THREAD

**Issue by ved1beta** [axolotl-ai-cloud/axolotl#3301]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

difference should be less or same

### Current behaviour

<img width="1107" height="523" alt="Image" src="https://github.com/user-attachments/assets/64dd0152-6a2c-47da-bc22-e16f4105a0f2" />

red is sample packing false

### Steps to reproduce

run config https://discord.com/channels/1104757954588196865/1111279858136383509/1444067990080589995

### Config yaml

```yaml
# Weights and Biases logging config
wandb_project: omlo-3-issue
wandb_name: vedaya-self
wandb_entity:
wandb_watch:
wandb_log_model:

# Model checkpointing config
output_dir: ./Outputs/BS1GA8-4K-Packing-On
save_steps: 100
save_safetensors: true
save_total_limit: 10
save_only_model: false

# Model architecture config
base_model: allenai/Olmo-3-7B-Instruct
revision_of_model:
model_type: AutoModelForCausalLM
tokenizer_type: AutoTokenizer
chat_template_jinja:

# Mixed precision training config
bf16: true
fp16: false
tf32: false

# Model loading config
lora_on_cpu: false
low_cpu_mem_usage: true
strict: false
load_in_8bit: false
load_in_4bit: true
bnb_config_kwargs:
  bnb_4bit_quant_storage: uint8

# Sequence config
sequence_len: 4096
min_sample_len: 128
sample_packing: false
eval_sample_packing: false
batch_flattening:
pad_to_sequence_len: true
train_on_inputs: false
group_by_length: false

# LoRA adapter config
peft_init_lora_weights:
adapter: qlora
lora_r: 32
lora_alpha: 32
peft_use_rslora:
lora_dropout: 0
lora_target_linear: true
lora_target_modules:
lora_target_parameters:
lora_modules_to_save:
lora_mlp_kernel:
lora_qkv_kernel:
lora_o_kernel:

# Dataset config
dataset_num_proc:
datasets:
  - path: PJMixers-Dev/allenai_tulu-3-sft-olmo-2-mixture-0225-filtered-ShareGPT
    split: train[:10000]
    type: chat_template
    field_messages: conversations
    message_property_mappings:
      role: from
      content: value
test_datasets:
val_set_size: 0
eval_strategy:
eval_steps:
evals_per_epoch:
dataset_prepared_path: ./00-Tokenized-Datasets/test-seed42
shuffle_before_merging_datasets: true
shuffle_merged_datasets: true
dataset_exact_deduplication:

# Training hyperparameters
max_steps: 500
num_epochs: 1
gradient_accumulation_steps: 8
micro_batch_size: 1
eval_batch_size: 1
warmup_steps: 10
optimizer: came_pytorch
optim_args:
lr_scheduler: rex
learning_rate: 1e-4
cosine_min_lr_ratio: 0.01
weight_decay: 0.01
max_grad_norm: 0.5
logging_steps: 1

# Model optimization
embeddings_skip_upcast: true
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false  # not sure what to use. maybe false is correct
activation_offloading:
sdp_attention: true
plugins:
  - axolotl.integrations.liger.LigerPlugin
  - axolotl.integrations.cut_cross_entropy.CutCrossEntropyPlugin
cut_cross_entropy: true
liger_rope: true
liger_rms_norm: true
liger_layer_norm: true
liger_glu_activation: true               # MLP LoRA Kernel overrides this
liger_cross_entropy: false               # Cut Cross Entropy overrides this
liger_fused_linear_cross_entropy: false  # Cut Cross Entropy overrides this
tiled_mlp:
tiled_mlp_use_original_mlp:
tiled_mlp_num_shards:

# Garbage Collection
gc_steps: 1

# Debug config
debug: true
seed: 42

# Token config
added_tokens_overrides:
special_tokens:
tokens:

lora_mlp_kernel: false
lora_qkv_kernel: false
lora_o_kernel: false
```

### Possible solution

NA

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.12

### axolotl branch-commit

main

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by ved1beta** [comment]
ig it worked : ) 
<img width="1830" height="844" alt="Image" src="https://github.com/user-attachments/assets/40ffa5f0-271c-4959-aff1-ad9e6cb0998a" />

**Comment by ved1beta** [comment]
was able to repro for only olmo model as of now
EDIT : difference exist for flex attention and sdpa with olmo models ( works fine for llama's )

**Comment by NanoCode012** [comment]
Closed by linked PR
